pkg/controller/master/upgrade/version_syncer.go

```diff
@@ -158,6 +159,9 @@ func (s *versionSyncer) syncVersions(resp CheckUpgradeResponse, currentVersion s
 	for _, v := range resp.Versions {
 		newVersion, err := s.getNewVersion(v)
 		if err != nil {
+			if strings.Contains(err.Error(), "failed to download version.yaml") {
```

Comment:
This will match any HTTP failure code (`resp.StatusCode < http.StatusOK || resp.StatusCode >= http.StatusBadRequest`), so it will just skip over things like 500 internal server error. Is that too loose, i.e. should we only be explicitly skipping 404s? Or am I being too picky? :-)

Reply:
I do not have a strong opinion about this. If we error out the sync job fails and we retry again after the sync interval. In this scenario we just keep trying to find the next version and hopefully if the error resolves itself the next sync interval will pick it up.

Reply:
Cool, sounds fine to me.

Reply:
If we have multiple upgrade versions, how do we make sure the upgrade version is we wanted if we do not error out here?

e.g. We have two versions, A and B. The A is newer than the B. But we got `failed to download version.yaml` with A, we chose B to upgrade. Is that what we want, or would this case not happen?

Reply:
Do we also need to consider the case where the error message is `"failed to download version-arm64.yaml"`? I feel the same way as Tim; we only need to skip 4xx errors. For others, it might be a server-level issue, so there is no need to try other versions quickly.

Reply:
currently we do not retry immediately when there is an error, the default behaviour is to log the error and retry in an hour, so this behaviour would continue as is

```
func (s *versionSyncer) start() {
	ticker := time.NewTicker(syncInterval)
	for {
		select {
		case <-ticker.C:
			if err := s.sync(); err != nil {
				logrus.Warnf("failed syncing upgrade versions: %v", err)
			}
		case <-s.ctx.Done():
			ticker.Stop()
			return
		}
	}
}
```

Reply:
i should probably just change the message to `failed to download version` so it is a valid substring for both arm64 and amd64 arch's

Reply:
I'm just being picky 😆
The updated version looks fine to me.